### PR TITLE
[MM-822]: Added testcases for http.go

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**
### Summary
Added testcases for http.go

#### Note
Unable to add testcase `completeUserOauth` function after the [line](https://github.com/mattermost-community/mattermost-plugin-msteams-meetings/blob/5121affd95c4596f58050f9d927b1dbc142b51f9/server/http.go#L130) as we were unable to mock the external API call made by this line's function.

### CODE REVIEW

File: server/http_test.go [hunk 1]
```diff
@@ -0,0 +1,609 @@
...
+	msgraph "github.com/yaegashi/msgraph.go/beta"
+)
+
+func TestConnectUser(t *testing.T) {
+	api := &plugintest.API{}
+	p := &Plugin{
+		MattermostPlugin: plugin.MattermostPlugin{
+			API: api,
+		},
+	}
```
Comment (raghavaggarwal2308) [inline server/http_test.go:26]: Should not be global

File: server/http_test.go [hunk 2]
```diff
@@ -0,0 +1,609 @@
...
+		expectedRedirectURL string
+		setup               func()
+	}{
+		{
+			name:                "Unauthorized User",
+			userID:              "",
+			channelID:           "testChannelID",
+			expectedStatus:      http.StatusUnauthorized,
+			expectedBody:        "Not authorized\n",
+			redirectExpected:    false,
```
Comment (raghavaggarwal2308) [inline server/http_test.go]: remove, check everywhere

File: server/http_test.go [hunk 3]
```diff
@@ -0,0 +1,609 @@
...
+			expectedRedirectURL: "",
+			setup: func() {
+				api.On("LogError", "connectUser, missing channelID in query params").Return(nil)
+			},
+		},
+		{
+			name:                "Error getting OAuth Config",
+			userID:              "testUserID",
+			channelID:           "testChannelID",
```
Comment (raghavaggarwal2308) [inline server/http_test.go]: Never changing, can remove

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

File: server/http_test.go (hunk 3 above)
Comment (Kshitij-Katiyar) [reply server/http_test.go]: it is needed for the line
`req := httptest.NewRequest(http.MethodGet, "/oauth2/connect?channelID="+tt.channelID, nil)`
for the testcase
`name:                "Missing Channel ID",`

Comment (Kshitij-Katiyar) [reply server/http_test.go]: same for user id

File: server/http_test.go [hunk 4]
```diff
@@ -36,37 +36,29 @@ func TestConnectUser(t *testing.T) {
...
-			channelID:           "testChannelID",
-			expectedStatus:      http.StatusInternalServerError,
-			expectedBody:        "error fetching siteUrl\n",
-			redirectExpected:    false,
-			expectedRedirectURL: "",
+			name:           "Error getting OAuth Config",
+			userID:         "testUserID",
+			channelID:      "testChannelID",
+			expectedStatus: http.StatusInternalServerError,
+			expectedBody:   "error fetching siteUrl\n",
```
Comment (raghavaggarwal2308) [inline server/http_test.go]:
```suggestion
			expectedBody:   "error fetching siteURL\n",
```

File: server/command_test.go [hunk 5]
```diff
@@ -427,7 +426,7 @@ func TestExecuteCommand(t *testing.T) {
 
 			mockAPI.On("SendEphemeralPost", tt.commandArgs.UserId, post).Return(&model.Post{}).Once()
 
-			response, _ := p.ExecuteCommand(&dummyPluginContext, &tt.commandArgs)
+			response, _ := p.ExecuteCommand(&dummyPluginContext, &tt.commandArgs) // #nosec G601
```
Comment (raghavaggarwal2308) [inline server/command_test.go]: Why have we added nosec

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

**Review by raghavaggarwal2308 — APPROVED** [review]

**Comment by Kshitij-Katiyar** [comment]
Created a PR on mattermost remote with all these changes hence closing this one

**Pull request closed without merging**